query/lookup.js

```diff
@@ -42,48 +41,39 @@ module.exports = function( db, names, points, cb ){
   // create prepared statement if one doesn't exist
   if( !stmt.hasOwnProperty( hash ) ){
 
-    // use named parameters to avoid sending coordinates twice for rtree conditions
-    var position = 1;
-
     // add point confitions to query
-    var pointConditions = Array.apply(null, new Array(max.points)).map(function(){
-      return POINT_SQL.replace('?A', '?' + position)
-                      .replace('?B', '?' + position++)
-                      .replace('?C', '?' + position)
-                      .replace('?D', '?' + position++);
+    var pointConditions = Array.apply(null, new Array(max.points)).map(function(__, i){
+      return POINT_SQL.replace(/\$lon/g, `$point${i}x`)
+                      .replace(/\$lat/g, `$point${i}y`);
```

Comment:
how about using lodash to make this more readable/expressive - _.times(max.points, (i) => {
...
})

Reply:
This is exactly what I was looking for any couldn't find!